core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -473,10 +473,26 @@ public boolean delete(Object key) {
          resizeLock.readLock().unlock();
       }
 
+      org.infinispan.commons.io.ByteBuffer keyBb;
+      org.infinispan.commons.io.ByteBuffer valueBb = null;
+      org.infinispan.commons.io.ByteBuffer metadataBb = null;
+
+      // If we only require the key, then no need to read disk
+      if (!loadValue && !loadMetadata) {
+         try {
+            keyBb = ctx.getMarshaller().objectToBuffer(key);
+            return ctx.getMarshalledEntryFactory().newMarshalledEntry(keyBb, valueBb, metadataBb);
```

Comment:
We should use https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/marshall/core/MarshalledEntryFactory.java#L18 so we don't have to marshall the key unless the caller wants the marshalled form.